pyspectral/atm_correction_ir.py

```diff
@@ -128,12 +129,29 @@ def delta(z):
         DELTA_REF = 6.2
         return (1 + DELTA_REF)**ratio(z, Z_0, Z_REF) - 1
 
-    y0, x0 = np.ma.where(view_zen == 0)
-    data[y0, x0] += tau0(data[y0, x0])
+    if hasattr(data, 'compute') or hasattr(view_zen, 'compute'):
+        is_dask_data = True
+    else:
+        is_dask_data = False
 
-    y, x = np.ma.where((view_zen > 0) & (view_zen < 90) & (~data.mask))
-    data[y, x] += tau(data[y, x]) * delta(view_zen[y, x])
-    return data
+    if is_dask_data:
+        dask_data = da.where(view_zen == 0,
+                             data + tau0(data),
+                             da.where((view_zen > 0) & (view_zen < 90),
+                                      data + (tau(data) * delta(view_zen)),
+                                      data))
```

Comment:
Could you split this into multiple lines. It is kind of complex for a single call and a little hard to read.

Reply:
done